<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1141</width>
    <height>852</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Forest Generator</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="s_mainWindowGridLayout">
    <item row="0" column="3">
     <widget class="QFrame" name="frame">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="frameShape">
       <enum>QFrame::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Raised</enum>
      </property>
     </widget>
    </item>
    <item row="0" column="0">
     <spacer name="horizontalSpacer">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <property name="sizeType">
       <enum>QSizePolicy::MinimumExpanding</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>765</width>
        <height>527</height>
       </size>
      </property>
     </spacer>
    </item>
    <item row="0" column="2">
     <widget class="QWidget" name="widget" native="true">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>330</width>
        <height>780</height>
       </size>
      </property>
      <widget class="QTabWidget" name="m_superTab">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>0</y>
         <width>320</width>
         <height>751</height>
        </rect>
       </property>
       <property name="sizePolicy">
        <sizepolicy hsizetype="Fixed" vsizetype="MinimumExpanding">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="minimumSize">
        <size>
         <width>320</width>
         <height>688</height>
        </size>
       </property>
       <property name="tabPosition">
        <enum>QTabWidget::North</enum>
       </property>
       <property name="tabShape">
        <enum>QTabWidget::Rounded</enum>
       </property>
       <property name="currentIndex">
        <number>0</number>
       </property>
       <widget class="QWidget" name="tab_3">
        <attribute name="title">
         <string>L-Systems</string>
        </attribute>
        <widget class="QTabWidget" name="m_tab">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>10</y>
           <width>300</width>
           <height>701</height>
          </rect>
         </property>
         <property name="sizePolicy">
          <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>300</width>
           <height>300</height>
          </size>
         </property>
         <property name="currentIndex">
          <number>0</number>
         </property>
         <widget class="QWidget" name="tab">
          <attribute name="title">
           <string>Tree1</string>
          </attribute>
          <widget class="QGroupBox" name="s_transformGB">
           <property name="geometry">
            <rect>
             <x>10</x>
             <y>10</y>
             <width>281</width>
             <height>651</height>
            </rect>
           </property>
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="title">
            <string>L-System Generation</string>
           </property>
           <layout class="QGridLayout" name="gridLayout">
            <item row="27" column="0">
             <layout class="QHBoxLayout" name="horizontalLayout_11">
              <item>
               <widget class="QCheckBox" name="m_seedToggle_1">
                <property name="text">
                 <string>Seed</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QSpinBox" name="m_seed_1">
                <property name="enabled">
                 <bool>false</bool>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item row="0" column="0">
             <widget class="QLabel" name="label">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="text">
               <string>Axiom</string>
              </property>
             </widget>
            </item>
            <item row="1" column="0">
             <widget class="QLineEdit" name="m_axiom_1">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="text">
               <string>FFFA</string>
              </property>
              <property name="maxLength">
               <number>32765</number>
              </property>
             </widget>
            </item>
            <item row="30" column="0">
             <widget class="QPushButton" name="m_resetCamera_1">
              <property name="text">
               <string>Reset Camera</string>
              </property>
             </widget>
            </item>
            <item row="24" column="0">
             <layout class="QHBoxLayout" name="horizontalLayout_4">
              <item>
               <widget class="QLabel" name="label_7">
                <property name="text">
                 <string>Angle Scale</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QDoubleSpinBox" name="m_angleScale_1">
                <property name="singleStep">
                 <double>0.010000000000000</double>
                </property>
                <property name="value">
                 <double>0.900000000000000</double>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item row="29" column="0">
             <widget class="QPushButton" name="m_generate_1">
              <property name="text">
               <string>Generate</string>
              </property>
             </widget>
            </item>
            <item row="17" column="0">
             <layout class="QHBoxLayout" name="horizontalLayout">
              <item>
               <widget class="QLabel" name="label_4">
                <property name="text">
                 <string>Step Size</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QDoubleSpinBox" name="m_stepSize_1">
                <property name="decimals">
                 <number>1</number>
                </property>
                <property name="singleStep">
                 <double>0.100000000000000</double>
                </property>
                <property name="value">
                 <double>2.000000000000000</double>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item row="18" column="0">
             <layout class="QHBoxLayout" name="horizontalLayout_2">
              <item>
               <widget class="QLabel" name="label_5">
                <property name="text">
                 <string>Step Scale</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QDoubleSpinBox" name="m_stepScale_1">
                <property name="singleStep">
                 <double>0.010000000000000</double>
                </property>
                <property name="value">
                 <double>0.900000000000000</double>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item row="21" column="0">
             <layout class="QHBoxLayout" name="horizontalLayout_3">
              <item>
               <widget class="QLabel" name="label_6">
                <property name="text">
                 <string>Angle</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QDoubleSpinBox" name="m_angle_1">
                <property name="decimals">
                 <number>1</number>
                </property>
                <property name="value">
                 <double>30.000000000000000</double>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item row="26" column="0">
             <layout class="QHBoxLayout" name="horizontalLayout_5">
              <item>
               <widget class="QLabel" name="label_3">
                <property name="text">
                 <string>Generations</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QSpinBox" name="m_generation_1">
                <property name="value">
                 <number>4</number>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item row="7" column="0">
             <widget class="QLineEdit" name="m_rule3_1">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
             </widget>
            </item>
            <item row="15" column="0">
             <widget class="QLineEdit" name="m_rule7_1">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
             </widget>
            </item>
            <item row="8" column="0">
             <widget class="QLineEdit" name="m_rule4_1">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
             </widget>
            </item>
            <item row="2" column="0">
             <widget class="QLabel" name="label_2">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="text">
               <string>Rules</string>
              </property>
             </widget>
            </item>
            <item row="28" column="0">
             <spacer name="verticalSpacer">
              <property name="orientation">
               <enum>Qt::Vertical</enum>
              </property>
              <property name="sizeType">
               <enum>QSizePolicy::Fixed</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>20</width>
                <height>5</height>
               </size>
              </property>
             </spacer>
            </item>
            <item row="3" column="0">
             <widget class="QLineEdit" name="m_rule1_1">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="text">
               <string>A=&quot;[B]////[B]////B</string>
              </property>
             </widget>
            </item>
            <item row="14" column="0">
             <widget class="QLineEdit" name="m_rule6_1">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
             </widget>
            </item>
            <item row="4" column="0">
             <widget class="QLineEdit" name="m_rule2_1">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="text">
               <string>B=&amp;FFFA</string>
              </property>
             </widget>
            </item>
            <item row="13" column="0">
             <widget class="QLineEdit" name="m_rule5_1">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </widget>
         <widget class="QWidget" name="tab_2">
          <attribute name="title">
           <string>Tree2</string>
          </attribute>
          <widget class="QGroupBox" name="s_transformGB_2">
           <property name="geometry">
            <rect>
             <x>10</x>
             <y>10</y>
             <width>281</width>
             <height>651</height>
            </rect>
           </property>
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="title">
            <string>L-System Generation</string>
           </property>
           <layout class="QGridLayout" name="gridLayout_2">
            <item row="2" column="0">
             <widget class="QLabel" name="label_14">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="text">
               <string>Rules</string>
              </property>
             </widget>
            </item>
            <item row="24" column="0">
             <widget class="QPushButton" name="m_resetCamera_2">
              <property name="text">
               <string>Reset Camera</string>
              </property>
             </widget>
            </item>
            <item row="23" column="0">
             <widget class="QPushButton" name="m_generate_2">
              <property name="text">
               <string>Generate</string>
              </property>
             </widget>
            </item>
            <item row="11" column="0">
             <layout class="QHBoxLayout" name="horizontalLayout_7">
              <item>
               <widget class="QLabel" name="label_9">
                <property name="text">
                 <string>Step Size</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QDoubleSpinBox" name="m_stepSize_2">
                <property name="decimals">
                 <number>1</number>
                </property>
                <property name="singleStep">
                 <double>0.100000000000000</double>
                </property>
                <property name="value">
                 <double>1.000000000000000</double>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item row="18" column="0">
             <layout class="QHBoxLayout" name="horizontalLayout_6">
              <item>
               <widget class="QLabel" name="label_8">
                <property name="text">
                 <string>Angle Scale</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QDoubleSpinBox" name="m_angleScale_2">
                <property name="singleStep">
                 <double>0.010000000000000</double>
                </property>
                <property name="value">
                 <double>0.900000000000000</double>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item row="7" column="0">
             <widget class="QLineEdit" name="m_rule5_2">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
             </widget>
            </item>
            <item row="15" column="0">
             <layout class="QHBoxLayout" name="horizontalLayout_8">
              <item>
               <widget class="QLabel" name="label_10">
                <property name="text">
                 <string>Angle</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QDoubleSpinBox" name="m_angle_2">
                <property name="decimals">
                 <number>1</number>
                </property>
                <property name="value">
                 <double>25.000000000000000</double>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item row="10" column="0">
             <widget class="QLineEdit" name="m_rule7_2">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
             </widget>
            </item>
            <item row="9" column="0">
             <widget class="QLineEdit" name="m_rule6_2">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
             </widget>
            </item>
            <item row="12" column="0">
             <layout class="QHBoxLayout" name="horizontalLayout_10">
              <item>
               <widget class="QLabel" name="label_12">
                <property name="text">
                 <string>Step Scale</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QDoubleSpinBox" name="m_stepScale_2">
                <property name="singleStep">
                 <double>0.010000000000000</double>
                </property>
                <property name="value">
                 <double>0.900000000000000</double>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item row="20" column="0">
             <layout class="QHBoxLayout" name="horizontalLayout_9">
              <item>
               <widget class="QLabel" name="label_11">
                <property name="text">
                 <string>Generations</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QSpinBox" name="m_generation_2">
                <property name="value">
                 <number>6</number>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item row="6" column="0">
             <widget class="QLineEdit" name="m_rule4_2">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="text">
               <string/>
              </property>
             </widget>
            </item>
            <item row="0" column="0">
             <widget class="QLabel" name="label_13">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="text">
               <string>Axiom</string>
              </property>
             </widget>
            </item>
            <item row="5" column="0">
             <widget class="QLineEdit" name="m_rule3_2">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
             </widget>
            </item>
            <item row="4" column="0">
             <widget class="QLineEdit" name="m_rule2_2">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="text">
               <string>F=FF</string>
              </property>
             </widget>
            </item>
            <item row="3" column="0">
             <widget class="QLineEdit" name="m_rule1_2">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="text">
               <string>A=F&amp;[[A]^A]^F^[^FA]&amp;A</string>
              </property>
             </widget>
            </item>
            <item row="1" column="0">
             <widget class="QLineEdit" name="m_axiom_2">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="text">
               <string>///A</string>
              </property>
              <property name="maxLength">
               <number>32765</number>
              </property>
             </widget>
            </item>
            <item row="21" column="0">
             <layout class="QHBoxLayout" name="horizontalLayout_12">
              <item>
               <widget class="QCheckBox" name="m_seedToggle_2">
                <property name="text">
                 <string>Seed</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QSpinBox" name="m_seed_2">
                <property name="enabled">
                 <bool>false</bool>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item row="22" column="0">
             <spacer name="verticalSpacer_2">
              <property name="orientation">
               <enum>Qt::Vertical</enum>
              </property>
              <property name="sizeType">
               <enum>QSizePolicy::Fixed</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>20</width>
                <height>5</height>
               </size>
              </property>
             </spacer>
            </item>
           </layout>
          </widget>
         </widget>
        </widget>
       </widget>
       <widget class="QWidget" name="tab_4">
        <attribute name="title">
         <string>Layout</string>
        </attribute>
        <widget class="QPushButton" name="m_resetCamera_layout">
         <property name="geometry">
          <rect>
           <x>30</x>
           <y>20</y>
           <width>257</width>
           <height>25</height>
          </rect>
         </property>
         <property name="text">
          <string>Reset Camera</string>
         </property>
        </widget>
       </widget>
       <widget class="QWidget" name="tab_5">
        <attribute name="title">
         <string>Render</string>
        </attribute>
        <widget class="QPushButton" name="m_resetCamera_render">
         <property name="geometry">
          <rect>
           <x>30</x>
           <y>20</y>
           <width>257</width>
           <height>25</height>
          </rect>
         </property>
         <property name="text">
          <string>Reset Camera</string>
         </property>
        </widget>
       </widget>
      </widget>
      <widget class="QCheckBox" name="m_grid">
       <property name="geometry">
        <rect>
         <x>230</x>
         <y>750</y>
         <width>91</width>
         <height>26</height>
        </rect>
       </property>
       <property name="text">
        <string>Show Grid</string>
       </property>
       <property name="checked">
        <bool>true</bool>
       </property>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1141</width>
     <height>25</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
